nestkernel/conn_builder.cpp

```diff
@@ -423,6 +551,185 @@ nest::OneToOneBuilder::connect_()
   }
 }
 
+/**
+ * Solves the disconnection of two nodes on a OneToOne basis without
+ * structural plasticity. This means this method can be manually called
+ * by the user to delete existing synapses.
+ */
+void
+nest::OneToOneBuilder::disconnect_()
+{
+  // make sure that target and source population have the same size
+  if ( sources_.size() != targets_.size() )
+  {
+    net_.message( SLIInterpreter::M_ERROR,
+      "Disconnect",
+      "Source and Target population must be of the same size." );
+    throw DimensionMismatch();
+  }
+
+#pragma omp parallel
+  {
+    // get thread id
+    const int tid = net_.get_thread_id();
+
+    try
+    {
+      // allocate pointer to thread specific random generator
+      librandom::RngPtr rng = net_.get_rng( tid );
+
+      for ( GIDCollection::const_iterator tgid = targets_.begin(), sgid = sources_.begin();
+            tgid != targets_.end();
+            ++tgid, ++sgid )
+      {
+
+        assert( sgid != sources_.end() );
+
+        if ( *sgid == *tgid and not autapses_ )
+          continue;
```

Comment:
Does `autapses_` (and `multapses_`) make sense during disconnect? I think not and would remove the skipping.